include/NovelRT.Interop/Animation/NrtSpriteAnimatorState.h

```diff
@@ -0,0 +1,32 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_INTEROP_ANIMATION_SPRITEANIMATORSTATE_H
+#define NOVELRT_INTEROP_ANIMATION_SPRITEANIMATORSTATE_H
+
+#include <stddef.h>
+#include "NovelRT.Interop/Animation/NrtAnimationTypedefs.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+NrtSpriteAnimatorState Nrt_SpriteAnimatorState_create();
+NrtResult Nrt_SpriteAnimatorState_insertNewState(NrtSpriteAnimatorState state, NrtSpriteAnimatorState stateTarget, NrtSpriteAnimatorStateConditionFunctions vector);
+NrtResult Nrt_SpriteAnimatorState_removeStateAtIndex(NrtSpriteAnimatorState state, size_t index);
+int32_t Nrt_SpriteAnimatorState_getShouldLoop(NrtSpriteAnimatorState state);
```

Comment:
This return type should be `NrtBool` no?

Reply:
Nope - these are void returns in C++ and do not return a status.

If you want, I _could_ make these void, but I figured since it's a complex operation we'd want to return success or a failure reason.